src/Main.hs

```diff
@@ -236,12 +236,26 @@ main =
 
       setProfile direction True profileText pathText
 
-      -- Switch to the new system configuration
-      liftIO (Control.Exception.catch switchSystem (errorHandler [Neat.text|
-[x] Failed to switch $targetText to $pathText
+      let successMsg = [Neat.text|[+] Succeeded switching $targetText to $pathText|]
+
+      switchSystem >>= \case
+        ExitSuccess -> stderrLines successMsg
+
+        -- This is the exit code returned by switch-to-configuration
+        -- if the configuration is successfully switched-to (using
+        -- `--switch`) but a service failed to start or restart during
+        -- the switch. We want to treat this as success but should
+        -- tell the user what happened...
+        ExitFailure 4 ->
+          stderrLines [Neat.text|
```

Comment:
Maybe provide a `--fail-on-partial-success` or somesuch so that a user can designate this as a failure?